File: MajorityElement/MajorityElement.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        major = 0
        counts = 0
       	for i in xrange(len(nums)):
       		if counts == 0:
       			major = nums[i]
       			counts += 1
       		else:
       			if nums[i] == major:
       				counts += 1
       			else:
       				counts -= 1
       	return major

File: MajorityElement/I/MajorityElement.cpp
#include<vector>
#include<algorithm>
using namespace std;

class Solution {
public:
	int majorityElement(vector<int>& nums) {
		int major, counts = 0, len = nums.size();
		for (int i = 0; i < len; i++) {
			if (!counts)	{
				major = nums[i];
				counts += 1;
			}
			else {
				counts += (nums[i] == major) ? 1 : -1;
			}
		}
		return major;
	}
};